rflx/model/model.py

```diff
@@ -59,8 +59,13 @@ def write_specification_files(self, output_dir: Path) -> None:
         for package, specification in self.create_specifications().items():
             (output_dir / f"{package.flat.lower()}.rflx").write_text(specification)
 
-    def _validate(self) -> None:
+    def _add_missing_types_and_validate(self) -> None:
         error = self._check_duplicates()
+
+        types = [t for s in self._sessions for t in s.dependencies.values()] + [*self._types]
+        types = [d for t in types for d in t.dependencies]
+        self._types = list(unique(types))
```

Comment:
Looking at this code, I realized that the semantics of `Session.dependencies` is inconsistent to `Type.dependencies`. While the latter determines the dependencies recursively, the former just returns the direct dependencies and thus corresponds to `Type.direct_dependencies`. So `Session.dependencies` should be renamed to `Session.direct_dependencies` to be consistent.

Reply:
Done.